### DISCUSSION TITLE
Write stream chunks directly to the output stream

### DISCUSSION THREAD

**Discussion by mokhaled2992** [hyperium/hyper#3721] [Q&A]
I asked [this question](https://github.com/tokio-rs/axum/discussions/2854) in the axum discussions section and I was referred here. I want to stream byte chunks to the output stream directly without having to yield those chunks in separate allocations. Perhaps there is a way to get some handle to the output stream and write the chunks directly to it? Is such a thing possible?

**Comment by seanmonstar** [comment]
There is no current way to do that, and likely won't be added. Granted, hyper doesn't require you provide the `Bytes` type itself, but really any owned type that implements `Buf`. So if you have a more optimal way you wish to store the data, you could do that.

If you're writing data to a `Vec<u8>`, you _could_ instead push it into a `BytesMut` and then slice off parts. That tries to reuse the allocation as much as possible.
